paimon-core/src/main/java/org/apache/paimon/rest/RESTCatalog.java

```diff
@@ -257,16 +270,133 @@ public void alterDatabase(String name, List<PropertyChange> changes, boolean ign
+            String databaseName, Integer maxResults, String pageToken)
+            throws DatabaseNotExistException {
+        Map<String, String> queryParams = Maps.newHashMap();
+        if (Objects.nonNull(maxResults) && maxResults > 0) {
+            queryParams.put(MAX_RESULTS, maxResults.toString());
+        }
+        if (Objects.nonNull(pageToken)) {
+            queryParams.put(PAGE_TOKEN, pageToken);
+        }
         try {
             ListTablesResponse response =
                     client.get(
                             resourcePaths.tables(databaseName),
+                            queryParams,
                             ListTablesResponse.class,
                             restAuthFunction);
-            if (response.getTables() != null) {
-                return response.getTables();
+            if (Objects.nonNull(response) && Objects.nonNull(response.getTables())) {
+                return new PagedList<>(response.getTables(), response.getNextPageToken());
+            } else {
+                LOG.warn(
+                        "response of listTablesPaged for {} is null with maxResults {} and pageToken {}",
+                        databaseName,
+                        maxResults,
+                        pageToken);
+                return new PagedList<>(Collections.emptyList(), null);
+            }
+        } catch (NoSuchResourceException e) {
+            throw new DatabaseNotExistException(databaseName);
+        }
+    }
+
+    @Override
+    public PagedList<Table> listTableDetailsPaged(
+            String databaseName, Integer maxResults, String pageToken)
+            throws DatabaseNotExistException {
+        Map<String, String> queryParams = Maps.newHashMap();
+        if (Objects.nonNull(maxResults) && maxResults > 0) {
+            queryParams.put(MAX_RESULTS, maxResults.toString());
+        }
+        if (Objects.nonNull(pageToken)) {
+            queryParams.put(PAGE_TOKEN, pageToken);
+        }
+        try {
+            ListTableDetailsResponse response =
+                    client.get(
+                            resourcePaths.tableDetails(databaseName),
+                            queryParams,
+                            ListTableDetailsResponse.class,
+                            restAuthFunction);
+            if (Objects.nonNull(response) && Objects.nonNull(response.getTableDetails())) {
+                return new PagedList<>(
+                        response.getTableDetails().stream()
+                                .map(
+                                        getTableResponse -> {
+                                            Identifier identifier =
+                                                    Identifier.create(
+                                                            databaseName,
+                                                            getTableResponse.getName());
+                                            try {
```

Comment:
Is it possible to reuse some utility methods, such as `getTable(Identifier identifier)`